### DISCUSSION TITLE
kbengine 如何调试python代码？？？？？？？？？？？？

### DISCUSSION THREAD

**Discussion by linxiubao** [kbengine/kbengine#1582] [Q&A]
kbengine 如何调试python代码？？？？？？？？？？？？

**Comment by kaisong524118** [comment]
可以使用断点python&nbsp; &nbsp;pydevd-pycharm.egg&nbsp;


isAddEnv = False
def SETTRACE(**args):
	host = args.get('host','localhost')
	port = args.get('port',777)
	global isAddEnv
	if not isAddEnv:
		if not KBEngine.hasRes('common/pydevd-pycharm.egg'): return
		fullpath = KBEngine.getResFullPath('common/pydevd-pycharm.egg')
		sys.path.append(fullpath)
		isAddEnv = True
	import pydevd_pycharm
	pydevd_pycharm.settrace(host, port=port, stdoutToServer=True, stderrToServer=True)





K.s
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "kbengine/kbengine"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年11月23日(星期三) 上午7:43
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;Re: [kbengine/kbengine] kbengine 如何调试python代码？？？？？？？？？？？？ (Discussion #1582)





 
https://blog.csdn.net/qq_37543025/article/details/86656981 不过不建议折腾,打log可以满足大部分情况
